blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,212 @@
+    url: https://github.com/sourcegraph/sourcegraph/pull/23697
+    category: 
+  - description: Results are now streamed from searcher by default, improving memory usage and latency for large, unindexed searches.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23754
+    category: Search
+  - description: "[`deploy-sourcegraph` overlays](https://docs.sourcegraph.com/admin/install/kubernetes/configure#overlays) now use `resources:` instead of the [deprecated `bases:` field] -(https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/bases/) for referencing Kustomize bases. deploy-sourcegraph#3606"
+    url: "https://github.com/sourcegraph/deploy-sourcegraph/pull/3606"
+    category: Admin
+  - description: "The `deploy-sourcegraph-docker` Pure Docker deployment scripts and configuration has been moved to the `./pure-docker` subdirectory. deploy-sourcegraph-docker#454"
+    url: "https://github.com/sourcegraph/deploy-sourcegraph-docker/pull/454"
+    category: 
+  - description: "In Kubernetes deployments, setting the `SRC_GIT_SERVERS` environment variable explicitly is no longer needed. Addresses of the gitserver pods will be discovered automatically and in the same numerical order as with the static list. Unset the env var in your `frontend.Deployment.yaml` to make use of this feature."
+    url: https://github.com/sourcegraph/sourcegraph/pull/24094
+    category: 
+  - description: "The consistent hashing scheme used to distribute repositories across indexed-search replicas has changed to improve distribution and reduce load discrepancies. In the next upgrade, indexed-search pods will re-index the majority of repositories since the repo to replica assignments will change. This can take a few hours in large instances, but searches should succeed during that time since a replica will only delete a repo once it has been indexed in the new replica that owns it. You can monitor this process in the Zoekt Index Server Grafana dashboard - the \"assigned\" repos in \"Total number of repos\" will spike and then reduce until it becomes the same as \"indexed\". As a fail-safe, the old consistent hashing scheme can be enabled by setting the `SRC_ENDPOINTS_CONSISTENT_HASH` env var to `consistent(crc32ieee)` in the `sourcegraph-frontend` deployment."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23921
+    category: Search
+  - description: "GitHub repository permissions can now leverage caching of team and organization permissions for user permissions syncing. Caching behaviour can be enabled via the `authorization.groupsCacheTTL` field in the code host config. This can significantly reduce the amount of time it takes to perform a full permissions sync due to reduced instances of being rate limited by the code host."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23978
+    category: Repositories
```

Comment:
```suggestion
  - description: "GitHub repository permissions can now leverage caching of team and organization permissions for user permissions syncing. This experimental caching behaviour can be enabled via the `authorization.groupsCacheTTL` field in the code host config. This can significantly reduce the amount of time it takes to perform a full permissions sync due to reduced instances of being rate limited by the code host for very large deployments. [Learn more](https://docs.sourcegraph.com/admin/repo/permissions#teams-and-organizations-permissions-caching)."
    url: https://github.com/sourcegraph/sourcegraph/pull/23978
    category: Repositories
```